From bb8dbe46e30ada4740c421c207cac419f661a062 Mon Sep 17 00:00:00 2001
From: Scott Murray <scott.murray@konsulko.com>
Date: Wed, 26 Nov 2025 23:04:45 -0500
Subject: [PATCH 3/4] Fix demo mode behavior

Add back the conditionals that prevent changing the speed and rpm
by touching the gauges when demo mode is on.  This avoids odd
jumps in the displayed values from user input in demo mode.

Bug-AGL: SPEC-xxxx

Signed-off-by: Scott Murray <scott.murray@konsulko.com>
---
 Widgets/ICPage.py | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/Widgets/ICPage.py b/Widgets/ICPage.py
index dea9191..144cb0a 100644
--- a/Widgets/ICPage.py
+++ b/Widgets/ICPage.py
@@ -294,11 +294,11 @@ class ICWidget(Base, Form):
             self.Speed_Gauge.rootObject().setProperty('animationDuration', 500)
 
         # Send updated speed to kuksa if simulator is not running
-        # if not self.simulator_running:
-        try:
-            threading.Thread(target=self.kuksa_client.set, args=(self.IC.speed, str(speed), 'value')).start()
-        except Exception as e:
-            logging.error(f"Error sending values to kuksa {e}")
+        if not self.simulator_running:
+            try:
+                threading.Thread(target=self.kuksa_client.set, args=(self.IC.speed, str(speed), 'value')).start()
+            except Exception as e:
+                logging.error(f"Error sending values to kuksa {e}")
 
     def update_RPM(self, source, value):
         """
@@ -330,11 +330,11 @@ class ICWidget(Base, Form):
             self.RPM_Gauge.rootObject().setProperty('animationDuration', 1000)
 
         # Send updated RPM to kuksa if simulator is not running
-        # if not self.simulator_running:
-        try:
-            threading.Thread(target=self.kuksa_client.set, args=(self.IC.engineRPM, str(rpm), 'value')).start()
-        except Exception as e:
-            logging.error(f"Error sending values to kuksa {e}")
+        if not self.simulator_running:
+            try:
+                threading.Thread(target=self.kuksa_client.set, args=(self.IC.engineRPM, str(rpm), 'value')).start()
+            except Exception as e:
+                logging.error(f"Error sending values to kuksa {e}")
 
     def update_coolantTemp(self, source, value):
         """
-- 
2.51.0

